### PR TITLE
Improve error message for sorting by child object

### PR DESCRIPTION
**Pull request by cmelchior**
Added unit test for capturing missing functionality + improved error message for bug reported here: 
https://github.com/realm/realm-java/issues/709

@bmunkholm @emanuelez @kneth

### CODE REVIEW

**Comment by kneth** [comment]
:+1:

File: realm/src/androidTest/java/io/realm/RealmResultsTest.java [hunk 1]
```diff
@@ -550,6 +553,27 @@ public void testSortWithTwoLanguages() {
...
+        RealmQuery<Owner> query = testRealm.where(Owner.class);
+        RealmResults<Owner> owners = query.findAll();
+
+        try {
+            owners.sort("cat.name");
+        } catch (IllegalArgumentException e) {
+            return; // Not supported yet
+        }
+
+        fail("Sorting by child object properties should result in a IllegalArgumentException");
```
Comment (emanuelez) [inline realm/src/androidTest/java/io/realm/RealmResultsTest.java]: If you put this fail() right after owners.sort() you won't need the return. And it's how we check for exceptions in the other tests

File: realm/src/main/java/io/realm/RealmResults.java [hunk 2]
```diff
@@ -186,14 +186,16 @@ public void sort(String fieldName, boolean sortAscending) {
         TableOrView table = getTable();
 
         if (table instanceof TableView) {
+            if (fieldName != null && fieldName.contains(".")) {
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/RealmResults.java]: We should check if fieldname is null every time, not only in this case.

**Comment by bmunkholm** [comment]
agree w. @emanuelez and then just update changelog.

**Pull request merged**